pygfx/renderers/wgpu/_pipeline.py

```diff
@@ -403,6 +405,8 @@ def update(self, wobject, environment, changed):
                     self.render_containers.append(RenderPipelineContainer(shader))
                 else:
                     raise ValueError(f"Shader type {shader.type} is unknown.")
+                if shader.needs_bake_function:
+                    self.bake_functions.append(shader.bake_function)
```

Comment:
is it possible somehow for this function to be appended more than once, accidentally? just wondering if such a codepath is possible and if it would create issues

Reply:
Not really. The `self.bake_functions` is reset above. But I understand the concern, since this path is quite critical. I refactored this code a bit to store it as a tuple instead, which makes it less likely its appended to unintentionally.